refactor: replace `boost::lexical_cast<std::string>` with `to_string`

mvadari opened the pull request:
## High Level Overview of Change

This PR does a bit of cleanup in the tests to replace some uses of `boost::lexical_cast<std::string>` with `to_string`.

### Context of Change

General cleanup/simplification

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

### API Impact

N/A

## Test Plan

CI passes.

bthomee commented on src/test/app/NFTokenBurn_test.cpp:
```diff
@@ -88,10 +88,8 @@ class NFTokenBurnBaseUtil_test : public beast::unit_test::suite
-            Json::Value jrr = env.rpc(
-                "json",
-                "ledger_data",
-                boost::lexical_cast<std::string>(jvParams));
```
Can the `#include <boost/algorithm/string/predicate.hpp>` now also be removed in these files?

mvadari replied:
None of them have that header (or still use it elsewhere)

bthomee replied:
I see it at least in the `NoRippleCheck_test.cpp`. But it looks like the `boost::lexical_cast` requires another include, so it must be a transitive dependency.

We'll separately have to make a pass over the codebase again to actually follow Include What You Use...

mvadari replied:
It's there, but there's some other boost string stuff still in that file, so I didn't want to remove it

bthomee committed:
refactor: replace boost::lexical_cast<std::string> with to_string

This change replaces boost::lexical_cast<std::string> with to_string in some of the tests to make them more readable.
